common/script/ops/readCard.js

```diff
@@ -1,10 +1,23 @@
-module.exports = function(user, req, cb) {
-  var cardType;
-  cardType = req.params.cardType;
-  user.items.special[cardType + "Received"].shift();
-  if (typeof user.markModified === "function") {
-    user.markModified("items.special." + cardType + "Received");
+import splitWhitespace from '../libs/splitWhitespace';
+import _ from 'lodash';
+import i18n from '../i18n';
+import {
+  BadRequest,
+} from '../libs/errors';
+
+module.exports = function readCard (user, req = {}) {
+  let cardType = _.get(req.params, 'cardType');
+
+  if (!cardType) {
+    throw new BadRequest(i18n.t('cardTypeRequired', req.language));
   }
+
+  user.items.special[`${cardType}Received`].shift();
+  user.markModified(`items.special.${cardType}Received`);
```

Comment:
markModified in this case was not necessary because mongoose will proxy the shift operation on the array, it was here before but it can be removed